File: Core/V1/Module/Github/Model/Repository.php
<?php

namespace Core\Github\Model;

class Repository
{
    private string $name;
    private ?string $description;
    private ?string $homepage;

    private bool $isPrivate;
    private bool $hasIssues;
    private bool $hasProjects;
    private bool $hasWiki;
    private bool $hasDownloads;
    private bool $isTemplate;
    private int $teamId;
    private bool $autoInit;

    private ?string $gitignoreTemplate;
    private ?string $licenseTemplate;

    private bool $allowSquashMerge;
    private bool $allowMergeCommit;
    private bool $allowMergeRebase;
    private bool $allowAutoMerge;
    private bool $deleteBranchOnMerge;
    private bool $useSquashPrTitleAsDefault;

    private ?string $squashMergeCommitTitle;
    private ?string $squashMergeCommitMessage;
    private ?string $mergeCommitTitle;
    private ?string $mergeCommitMessage;

    public function __construct(
        string $name,
        ?string $description,
        ?string $homepage,
        bool $isPrivate,
        bool $hasIssues,
        bool $hasProjects,
        bool $hasWiki,
        bool $hasDownloads,
        bool $isTemplate,
        int $teamId,
        bool $autoInit,
        ?string $gitignoreTemplate,
        ?string $licenseTemplate,
        bool $allowSquashMerge,
        bool $allowMergeCommit,
        bool $allowMergeRebase,
        bool $allowAutoMerge,
        bool $deleteBranchOnMerge,
        bool $useSquashPrTitleAsDefault,
        ?string $squashMergeCommitTitle,
        ?string $squashMergeCommitMessage,
        ?string $mergeCommitTitle,
        ?string $mergeCommitMessage
    ) {
        $this->name = $name;
        $this->description = $description;
        $this->homepage = $homepage;
        $this->isPrivate = $isPrivate;
        $this->hasIssues = $hasIssues;
        $this->hasProjects = $hasProjects;
        $this->hasWiki = $hasWiki;
        $this->hasDownloads = $hasDownloads;
        $this->isTemplate = $isTemplate;
        $this->teamId = $teamId;
        $this->autoInit = $autoInit;
        $this->gitignoreTemplate = $gitignoreTemplate;
        $this->licenseTemplate = $licenseTemplate;
        $this->allowSquashMerge = $allowSquashMerge;
        $this->allowMergeCommit = $allowMergeCommit;
        $this->allowMergeRebase = $allowMergeRebase;
        $this->allowAutoMerge = $allowAutoMerge;
        $this->deleteBranchOnMerge = $deleteBranchOnMerge;
        $this->useSquashPrTitleAsDefault = $useSquashPrTitleAsDefault;
        $this->squashMergeCommitTitle = $squashMergeCommitTitle;
        $this->squashMergeCommitMessage = $squashMergeCommitMessage;
        $this->mergeCommitTitle = $mergeCommitTitle;
        $this->mergeCommitMessage = $mergeCommitMessage;
    }

    public function serialize(): string
    {
        $jsonMap = [];

        $jsonMap['name'] = $this->name;

        if ($this->description !== null) {
            $jsonMap['description'] = $this->description;
        }

        if ($this->homepage !== null) {
            $jsonMap['homepage'] = $this->homepage;
        }

        if ($this->isPrivate) {
            $jsonMap['private'] = $this->isPrivate;
        }

        if ($this->hasIssues) {
            $jsonMap['has_issues'] = $this->hasIssues;
        }

        if ($this->hasProjects) {
            $jsonMap['has_projects'] = $this->hasProjects;
        }

        if ($this->hasWiki) {
            $jsonMap['has_wiki'] = $this->hasWiki;
        }

        if ($this->hasDownloads) {
            $jsonMap['has_downloads'] = $this->hasDownloads;
        }

        if ($this->isTemplate) {
            $jsonMap['is_template'] = $this->isTemplate;
        }

        if ($this->teamId !== -1) {
            $jsonMap['team_id'] = $this->teamId;
        }

        if ($this->autoInit) {
            $jsonMap['auto_init'] = $this->autoInit;
        }

        if ($this->gitignoreTemplate !== null) {
            $jsonMap['gitignore_template'] = $this->gitignoreTemplate;
        }

        if ($this->licenseTemplate !== null) {
            $jsonMap['license_template'] = $this->licenseTemplate;
        }

        if ($this->allowSquashMerge) {
            $jsonMap['allow_squash_merge'] = $this->allowSquashMerge;
        }

        if ($this->allowMergeCommit) {
            $jsonMap['allow_merge_commit'] = $this->allowMergeCommit;
        }

        if ($this->allowMergeRebase) {
            $jsonMap['allow_merge_rebase'] = $this->allowMergeRebase;
        }

        if ($this->allowAutoMerge) {
            $jsonMap['allow_auto_merge'] = $this->allowAutoMerge;
        }

        if ($this->deleteBranchOnMerge) {
            $jsonMap['delete_branch_on_merge'] = $this->deleteBranchOnMerge;
        }

        if ($this->useSquashPrTitleAsDefault) {
            $jsonMap['use_squash_pr_title_as_default'] = $this->useSquashPrTitleAsDefault;
        }

        if ($this->squashMergeCommitTitle !== null) {
            $jsonMap['squash_merge_commit_title'] = $this->squashMergeCommitTitle;
        }

        if ($this->squashMergeCommitMessage !== null) {
            $jsonMap['squash_merge_commit_message'] = $this->squashMergeCommitMessage;
        }

        if ($this->mergeCommitTitle !== null) {
            $jsonMap['merge_commit_title'] = $this->mergeCommitTitle;
        }

        if ($this->mergeCommitMessage !== null) {
            $jsonMap['merge_commit_message'] = $this->mergeCommitMessage;
        }

        return json_encode($jsonMap);
    }
}

File: Core/Lib/nanorest/src/main/java/org/sindria/xpipe/core/lib/nanorest/github/model/Repository.java
package org.sindria.xpipe.core.lib.nanorest.github.model;

import org.json.JSONObject;

import java.util.HashMap;
import java.util.Map;

public class Repository {

    private final String name;
    private final String description;
    private final String homepage;

    private final boolean isPrivate;

    private final boolean hasIssues;

    private final boolean hasProjects;

    private final boolean hasWiki;

    private final boolean hasDownloads;

    private final boolean isTemplate;

    private final int teamId;

    private final boolean autoInit;

    private final String gitignoreTemplate;
    private final String licenseTemplate;

    private final boolean allowSquashMerge;

    private final boolean allowMergeCommit;

    private final boolean allowMergeRebase;

    private final boolean allowAutoMerge;

    private final boolean deleteBranchOnMerge;

    private final boolean useSquashPrTitleAsDefault;

    private final String squashMergeCommitTitle;
    private final String squashMergeCommitMessage;
    private final String mergeCommitTitle;
    private final String mergeCommitMessage;


//        public Repository(String name) {
//        this.name = name;
//        this.description = "";
//        this.homepage = "";
//        this.isPrivate = false;
//        this.hasIssues = true;
//        this.hasProjects =  true;
//        this.hasWiki = true;
//        this.hasDownloads = true;
//        this.isTemplate = false;
//        this.teamId = 0;
//        this.autoInit = false;
//        this.gitignoreTemplate = "";
//        this.licenseTemplate = "";
//        this.allowSquashMerge = true;
//        this.allowMergeCommit = true;
//        this.allowMergeRebase = true;
//        this.allowAutoMerge = false;
//        this.deleteBranchOnMerge = false;
//        this.useSquashPrTitleAsDefault = false;
//        this.squashMergeCommitTitle = "";
//        this.squashMergeCommitMessage = "";
//        this.mergeCommitTitle = "";
//        this.mergeCommitMessage = "";
//    }

    public Repository(
            String name,
            String description,
            String homepage,
            boolean isPrivate,
            boolean hasIssues,
            boolean hasProjects,
            boolean hasWiki,
            boolean hasDownloads,
            boolean isTemplate,
            int teamId,
            boolean autoInit,
            String gitignoreTemplate,
            String licenseTemplate,
            boolean allowSquashMerge,
            boolean allowMergeCommit,
            boolean allowMergeRebase,
            boolean allowAutoMerge,
            boolean deleteBranchOnMerge,
            boolean useSquashPrTitleAsDefault,
            String squashMergeCommitTitle,
            String squashMergeCommitMessage,
            String mergeCommitTitle,
            String mergeCommitMessage
    ) {
        this.name = name;
        this.description = description;
        this.homepage = homepage;
        this.isPrivate = isPrivate;
        this.hasIssues = hasIssues;
        this.hasProjects = hasProjects;
        this.hasWiki = hasWiki;
        this.hasDownloads = hasDownloads;
        this.isTemplate = isTemplate;
        this.teamId = teamId;
        this.autoInit = autoInit;
        this.gitignoreTemplate = gitignoreTemplate;
        this.licenseTemplate = licenseTemplate;
        this.allowSquashMerge = allowSquashMerge;
        this.allowMergeCommit = allowMergeCommit;
        this.allowMergeRebase = allowMergeRebase;
        this.allowAutoMerge = allowAutoMerge;
        this.deleteBranchOnMerge = deleteBranchOnMerge;
        this.useSquashPrTitleAsDefault = useSquashPrTitleAsDefault;
        this.squashMergeCommitTitle = squashMergeCommitTitle;
        this.squashMergeCommitMessage = squashMergeCommitMessage;
        this.mergeCommitTitle = mergeCommitTitle;
        this.mergeCommitMessage = mergeCommitMessage;
    }

    public JSONObject serialize() {
        return new JSONObject(this.toJson());
    }

    private String toJson() {
        Map<String, Object> jsonMap = new HashMap<>();

        jsonMap.put("name", name);
        if (!(description == null)) jsonMap.put("description", description);
        if (!(homepage == null)) jsonMap.put("homepage", homepage);
        if (isPrivate) jsonMap.put("private", isPrivate);
        if (hasIssues) jsonMap.put("has_issues", hasIssues);
        if (hasProjects) jsonMap.put("has_projects", hasProjects);
        if (hasWiki) jsonMap.put("has_wiki", hasWiki);
        if (hasDownloads) jsonMap.put("has_downloads", hasDownloads);
        if (isTemplate) jsonMap.put("is_template", isTemplate);
        if (teamId != -1) jsonMap.put("team_id", teamId);
        if (autoInit) jsonMap.put("auto_init", autoInit);
        if (!(gitignoreTemplate == null)) jsonMap.put("gitignore_template", gitignoreTemplate);
        if (!(licenseTemplate == null)) jsonMap.put("license_template", licenseTemplate);
        if (allowSquashMerge) jsonMap.put("allow_squash_merge", allowSquashMerge);
        if (allowMergeCommit) jsonMap.put("allow_merge_commit", allowMergeCommit);
        if (allowMergeRebase) jsonMap.put("allow_merge_rebase", allowMergeRebase);
        if (allowAutoMerge) jsonMap.put("allow_auto_merge", allowAutoMerge);
        if (deleteBranchOnMerge) jsonMap.put("delete_branch_on_merge", deleteBranchOnMerge);
        if (useSquashPrTitleAsDefault) jsonMap.put("use_squash_pr_title_as_default", useSquashPrTitleAsDefault);
        if (!(squashMergeCommitTitle == null)) jsonMap.put("squash_merge_commit_title", squashMergeCommitTitle);
        if (!(squashMergeCommitMessage == null)) jsonMap.put("squash_merge_commit_message", squashMergeCommitMessage);
        if (!(mergeCommitTitle == null)) jsonMap.put("merge_commit_title", mergeCommitTitle);
        if (!(mergeCommitMessage == null)) jsonMap.put("merge_commit_message", mergeCommitMessage);


        return new JSONObject(jsonMap).toString();
    }
}
